uber/tests/models/test_badge_funcs.py

```diff
@@ -107,20 +107,21 @@ def test_preassigned_with_new(self, session):
         session.add(Attendee(badge_type=c.STAFF_BADGE, badge_num=6))
         assert 7 == session.get_next_badge_num(c.STAFF_BADGE)
 
+    def test_preassigned_two_new(self, session):
+        session.add(Attendee(badge_type=c.STAFF_BADGE, badge_num=6))
+        session.add(Attendee(badge_type=c.STAFF_BADGE, badge_num=7))
+        assert 8 == session.get_next_badge_num(c.STAFF_BADGE)
+
     def test_preassigned_with_dirty(self, session):
         session.supporter_five.badge_type, session.supporter_five.badge_num = c.STAFF_BADGE, 6
         assert 7 == session.get_next_badge_num(c.STAFF_BADGE)
         session.expunge(session.supporter_five)
 
-    def test_preassigned_with_new_gap(self, session):
```

Comment:
I removed these unit tests for two reasons:
1. The fix for #2436 breaks them (with no simple/sensible way to avoid that), and
2. They tested a condition that would never happen and, even if it did, we're not that concerned about it

We still 'fill in' gaps in badge numbers after they make it to the database -- we just don't attempt to fix gaps within a single session.